index.js

```diff
@@ -60,25 +62,56 @@ function WebpackGitHash(opts) {
 /**
  * Test if a file can be deleted, then delete it
  */
-WebpackGitHash.prototype.deleteObsoleteFile = function(filename) {
-  if ((this.regex.filename && this.regex.filename.test(filename)) ||
-    (this.regex.chunkFilename && this.regex.chunkFilename.test(filename))) {
-    // delete synchronously so we know when loopFiles() is complete
-    fs.unlinkSync(path.join(this.outputPath, filename));
-    console.log('Deleted ' + filename);
-    this.deletedFiles.push(filename);
+WebpackGitHash.prototype.deleteObsoleteFile = function(file) {
+  for (var i = 0; i < this.regex.length; i++) {
+    var currentRegex = this.regex[i];
+    var testPath = file.path.replace(this.outputPath, '');
+
+    if (currentRegex.test(testPath)) {
+      fs.unlink(file.path, function(err) {
+        if (err) {
+          console.log(err);
+        }
+        console.log('WebpackGitHash: Deleted ' + file.path);
+      });
+      this.deletedFiles.push(file.path);
+    }
   }
 }
 
 /**
- * Loop through files after reading folder contents
+ * Add a regex for a particular asset
  */
-WebpackGitHash.prototype.loopFiles = function(err, contents) {
-  if (err) {
-    throw err;
+WebpackGitHash.prototype.populateRegex = function(assetName) {
+  if (!this.regex.hasOwnProperty(assetName)) {
```

Comment:
hmm woops, forgot to change this, I might switch back to using an object for `this.regex` so we can cache the regex builds.